cli/compose/convert/volume.go

```diff
@@ -88,3 +67,77 @@ func convertVolumeToMount(
 	// Named volumes
 	return result, nil
 }
+
+func handleBindToMount(
+	volume composetypes.ServiceVolumeConfig,
+	result mount.Mount,
+) (mount.Mount, error) {
+	if volume.Volume != nil {
+		return result, errors.New("volume options are incompatible with type bind")
+	}
+	if volume.Tmpfs != nil {
+		return result, errors.New("tmpfs options are incompatible with type bind")
+	}
+	if volume.Bind != nil {
+		result.BindOptions = &mount.BindOptions{
+			Propagation: mount.Propagation(volume.Bind.Propagation),
+		}
+	}
+	// Binds volumes
+	return result, nil
+}
+
+func handleTmpfsToMount(
+	volume composetypes.ServiceVolumeConfig,
+	result mount.Mount,
+) (mount.Mount, error) {
+	// source must be empty for tmpfs
+	if volume.Source != "" {
+		return result, errors.New("invalid tmpfs source, source must be empty")
+	}
+	if volume.Bind != nil {
+		return result, errors.New("bind options are incompatible with type tmpfs")
+	}
+	if volume.Volume != nil {
+		return result, errors.New("volume options are incompatible with type tmpfs")
+	}
+	if volume.Tmpfs != nil {
+		result.TmpfsOptions = &mount.TmpfsOptions{
+			SizeBytes: volume.Tmpfs.Size,
+		}
+	}
+	// Tmpfs volumes
+	return result, nil
+}
+
+func convertVolumeToMount(
+	volume composetypes.ServiceVolumeConfig,
+	stackVolumes volumes,
+	namespace Namespace,
+) (mount.Mount, error) {
+	result := mount.Mount{
+		Type:        mount.Type(volume.Type),
+		Source:      volume.Source,
+		Target:      volume.Target,
+		ReadOnly:    volume.ReadOnly,
+		Consistency: mount.Consistency(volume.Consistency),
+	}
+
+	// Anonymous volumes
+	if volume.Source == "" && volume.Type != "tmpfs" {
+		return result, nil
+	}
+
+	switch volume.Type {
+	case "volume":
+		mount, err := handleVolumeToMount(volume, stackVolumes, namespace, result)
+		return mount, err
+	case "bind":
+		mount, err := handleBindToMount(volume, result)
+		return mount, err
+	case "tmpfs":
+		mount, err := handleTmpfsToMount(volume, result)
+		return mount, err
+	}
+	return result, nil
```

Comment:
This doesn't look right: unknown `volume.Type` should produce an error, and return an empty result